core/src/main/java/io/micronaut/core/io/service/MicronautMetaServiceLoaderUtils.java

```diff
@@ -257,7 +257,7 @@ public List<S> collect(boolean allowFork) {
                 List<S> collection = new ArrayList<>(serviceEntries.size());
                 for (String serviceEntry : serviceEntries) {
                     S val = instantiate(serviceEntry, classLoader);
-                    if (val != null && predicate != null && !predicate.test(val)) {
+                    if (val != null && !collection.contains(val) && (predicate == null || predicate.test(val))) {
```

Comment:
`collection.contains(val)` might be a costly operation, so I wonder if it is required. Do the services implement a correct equals method, so that this would actually make a difference?

If it is required,can we use a set instead of a list and don't do this check?

Reply:
The parallel option has it (line 312)
`if (result != null && !values.contains(result)) {`

That's why I added it.

Reply:
I think we an remove it there

Reply:
okak, I have committed the change